flatson/flatson.py

```diff
@@ -21,36 +21,56 @@ def serialization_options(self):
         return self.schema.get('flatson_serialize') or {}
 
 
-def create_getter(path, field_sep='.'):
-    if field_sep in path:
-        first_key, rest = path.split(field_sep, 1)
-        return lambda x: create_getter(rest)(x.get(first_key, {}))
-    else:
-        return lambda x: x.get(path, None)
-
-
-def infer_flattened_field_names(schema, field_sep='.'):
-    fields = []
+class FieldsInferrer(object):
+    def __init__(self, schema, field_sep='.'):
+        self.schema = schema
+        self.field_sep = field_sep
 
-    for key, value in schema.get('properties', {}).items():
-        val_type = value.get('type')
-        if val_type == 'object':
-            for subfield in infer_flattened_field_names(value):
-                full_name = '{prefix}{fsep}{extension}'.format(
-                    prefix=key, fsep=field_sep, extension=subfield.name)
-                fields.append(Field(full_name, create_getter(full_name), subfield.schema))
+    @property
+    def fields(self):
```

Comment:
This should be a method with name `infer`, not a property.

Also, this method is getting too big, we should refactor the insides of it too.